py2hwsw/scripts/iob_core.py

```diff
@@ -398,11 +398,29 @@ def update_global_top_module(self, attributes={}):
             self.set_default_attribute("build_dir", __class__.global_build_dir)
 
     attrs = [
-        "core_name",
-        "instance_name",
-        ["-p", "parameters", {"nargs": "+"}, "pairs"],
-        ["-c", "connect", {"nargs": "+"}, "pairs"],
-    ]
+            "core_name",
+		    "instance_name",
+		    ["-p", "parameters", {"nargs": "+"}, "pairs"],
+		    ["-c", "connect", {"nargs": "+"}, "pairs"],
+		    ["--no_autoaddr", "autoaddr", {"action": "store_false"}],
+		    ["--rw_overlap", "rw_overlap", {"action": "store_true"}],
+		    ["--csr_if", "csr_if"],
+		    {
+		        "--csr-group&csrs": [
+		            "name",
+		            {
+		                "-r&regs": [
+		                    "name:n_bits",
+                            ["-t", "type"],
+		                    ["--rst_val", "rst_val"],
+                            ["--addr", "addr", {"type":int}],
+		                    ["--log2n_items", "log2n_items"],
+		                    ["--no_autoreg", "autoreg", {"action": "store_false"}],
+		                ],
+		            },
+		        ]
+		    },
```

Comment:
Since these python parameters are specific to the `iob_csrs.py` core, I think it would be nice if they could be moved to the scripts of that core, instead of keeping them in the generic `iob_core.py` py2 script.

For example, the `autoaddr` is defined as a python parameter of the iob_csrs.py core [here](https://github.com/IObundle/py2hwsw/blob/7b3964632625d07fcf80acf453a956600bfa8298/py2hwsw/lib/hardware/iob_csrs/iob_csrs.py#L24).

However, the only way I see of doing this would be to pass the shorthand notation (as a string) as a python parameter to the `iob_csrs.py` core. And then re-implement the shorthand notation parser in the scripts of that core.

But for now, since the iob_csrs core is the only one currently using multi-level shorthand notation for python parameters, and the csrs are a common property of various cores, I also think its ok to keep the scripts where they are.